code/modules/codex/entries/_codex_entry.dm

```diff
@@ -33,48 +34,69 @@
 				PRINT_STACK_TRACE("Trying to save codex entry for [name] by path [associated_path] but one already exists!")
 			SScodex.entries_by_path[associated_path] = src
 
-	if(name)
-		var/clean_name = lowertext(sanitize(name))
-		LAZYDISTINCTADD(associated_strings, clean_name)
-	else if(length(associated_strings))
-		name = associated_strings[1]
-
-	if(length(associated_strings))
-
-		var/list/cleaned_strings = list()
-		for(var/associated_string in associated_strings)
-			cleaned_strings |= lowertext(trim(associated_string))
-		associated_strings = cleaned_strings
-
+	if(!name)
 		if(length(associated_strings))
-			for(var/key_string in associated_strings)
-				if(SScodex.entries_by_string[key_string])
-					PRINT_STACK_TRACE("Trying to save codex entry for [name] by string [key_string] but one already exists!")
-				SScodex.entries_by_string[key_string] = src
+			name = associated_strings[1]
+		else
+			CRASH("Attempted to instantiate unnamed codex entry with no associated strings!")
+
+	LAZYDISTINCTADD(associated_strings, codex_sanitize(name))
+	for(var/associated_string in associated_strings)
+		var/clean_string = codex_sanitize(associated_string)
+		if(!clean_string)
+			associated_strings -= associated_string
+			continue
+		if(clean_string != associated_string)
+			associated_strings -= associated_string
+			associated_strings |= clean_string
+		if(SScodex.entries_by_string[clean_string])
+			PRINT_STACK_TRACE("Trying to save codex entry for [name] by string [clean_string] but one already exists!")
+		SScodex.entries_by_string[clean_string] = src
 
 	..()
 
 /datum/codex_entry/Destroy(force)
 	SScodex.all_entries -= src
 	. = ..()
-	
-/datum/codex_entry/proc/get_header(var/mob/presenting_to)
-	var/list/dat = list()
-	var/datum/codex_entry/linked_entry = SScodex.get_entry_by_string("nexus")
-	dat += "<a href='?src=\ref[SScodex];show_examined_info=\ref[linked_entry];show_to=\ref[presenting_to]'>Home</a>"
-	dat += "<a href='?src=\ref[presenting_to.client];codex_search=1'>Search Codex</a>"
-	dat += "<a href='?src=\ref[presenting_to.client];codex_index=1'>List All Entries</a>"
-	dat += "<hr><h2>[name]</h2>"
-	return jointext(dat, null)
 
-/datum/codex_entry/proc/get_text(var/mob/presenting_to)
-	var/list/dat = list(get_header(presenting_to))
+/datum/codex_entry/proc/get_codex_header(var/mob/presenting_to)
+	. = list()
+	if(presenting_to)
+		var/datum/codex_entry/linked_entry = SScodex.get_entry_by_string("nexus")
+		. += "<a href='?src=\ref[SScodex];show_examined_info=\ref[linked_entry];show_to=\ref[presenting_to]'>Home</a>"
```

Comment:
Unless you are actually doing something nontrivial with `show_to`, I think it's better practice to infer it from `usr` in `Topic`.